src/main/java/com/org/jenkins/custom/jenkins/distribution/service/services/PackagerDownloadService.java

```diff
@@ -24,19 +22,18 @@
 
     public ResponseEntity<Resource> downloadWAR(String versionName, String configuration) throws Exception {
         File warFile = null;
-        String artifactId = getArtifactId();
         try {
             warFile = WarGenerator.generateWAR(versionName, configuration);
             InputStreamResource resource = new InputStreamResource(new FileInputStream(warFile));
             String headerValue = "attachment; filename=jenkins.war";
             LOGGER.info("Returning War file");
             return returnResource(returnHeaders(headerValue), warFile, resource);
         } catch (Exception e) {
-            e.printStackTrace();
+            throw e;
```

Comment:
Propagating the error upwards so that the file can be deleted if its exists and it is a good practise to let controllers handle errors.